sources/host-ctr/cmd/host-ctr/main_test.go

```diff
@@ -0,0 +1,161 @@
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "registry-1.docker.io",
+					Scheme:       "https",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+			},
+		},
+		{
+			"No scheme",
+			"docker.io",
+			RegistryConfig{
+				Mirrors: map[string]Mirror{
+					"docker.io": {
+						Endpoints: []string{"localhost", "198.158.0.0", "127.0.0.1"},
+					},
+				},
+			},
+			[]docker.RegistryHost{
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "localhost",
+					Scheme:       "http",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "198.158.0.0",
+					Scheme:       "https",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "127.0.0.1",
+					Scheme:       "http",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "registry-1.docker.io",
+					Scheme:       "https",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+			},
+		},
+		{
+			"* endpoints",
+			"weird.io",
+			RegistryConfig{
+				Mirrors: map[string]Mirror{
+					"docker.io": {
+						Endpoints: []string{"notme", "certainly-not-me"},
+					},
+					"*": {
+						Endpoints: []string{"198.158.0.0", "example.com"},
+					},
+				},
+			},
+			[]docker.RegistryHost{
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "198.158.0.0",
+					Scheme:       "https",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "example.com",
+					Scheme:       "https",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "weird.io",
+					Scheme:       "https",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+			},
+		},
+		{
+			"No mirrors",
+			"docker.io",
+			RegistryConfig{
+				Mirrors: map[string]Mirror{},
+			},
+			[]docker.RegistryHost{
+				{
+					Authorizer:   docker.NewDockerAuthorizer(),
+					Host:         "registry-1.docker.io",
+					Scheme:       "https",
+					Path:         "/v2",
+					Capabilities: docker.HostCapabilityResolve | docker.HostCapabilityPull,
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			result, err := _registryHosts(tc.host, &tc.config, docker.NewDockerAuthorizer())
```

Comment:
`_registryHosts` doesn't exist anymore.  Looks like we don't yet have automated `go test` runs as part of our build; we can tackle that in a separate PR if you'd prefer but let's make sure the tests that we're checking in run properly.